test/react-native/features/device.feature

```diff
@@ -0,0 +1,100 @@
+Feature: Device data
+
+Scenario: Handled JS error
+  When I run "DeviceJsHandledScenario"
+  Then I wait to receive a request
+  And the exception "errorClass" equals "Error"
+  And the exception "message" equals "DeviceJsHandledScenario"
+  And the event "unhandled" is false
+
+  # Android
+  And the event "device.id" is not null
+  And the event "device.jailbroken" is false
+  And the event "device.locale" equals "en_US"
+  And the event "device.manufacturer" is not null
+  And the event "device.model" is not null
+  And the event "device.osName" equals "android"
+  And the event "device.osVersion" matches "^\d+$"
+  And the event "device.runtimeVersions.androidApiLevel" is not null
+  And the event "device.runtimeVersions.reactNative" matches "^\d+\.\d+\.\d+$"
+  And the event "device.runtimeVersions.reactNativeJsEngine" matches "jsc"
+  And the event "device.totalMemory" is not null
```

Comment:
I imagine this is the best we can do currently, but I would be concerned about the zero case.  Perhaps we could raise an internal ticket to add a new step along the lines of "and the event {string} is a number greater than {int}" to tighten this up in the future.

Reply:
Yes, best we can do at present. I was on the cusp of making a variety of maze runner feature requests…

- `and the event {string} is an integer greater than {int}`
- `and the event {string} is an integer between {int} and {int}` (useful for `apiLevel`)
- `and the event {string} is a number` (general case where the number is allowed to be anything, but more type-guarantee than simply `not null`

Reply:
No problem, just send the ticket reference our way.